onnxmltools/convert/common/optimizer.py

```diff
@@ -35,7 +35,8 @@ def is_transpose(self):
 
     @property
     def in_single_path(self):
-        return len(self.successor) <= 1
+        return len(self.successor) == 1 and not self.successor[0].in_or_out and \
+               len(self.precedence) == 1 and len(self.precedence[0].successor) <= 1
 
     @property
     def in_or_out(self):
```

Comment:
I don't quite understand this attribute.

Reply:
Optimization could not happen on fan in/out node.